### PR TITLE
BLD: fix include list for sdist building

### PR DESCRIPTION
**Pull request by rgommers**
Closes gh-11927

To verify the changes I ran `python setup.py sdist` on `master` and on the branch this PR was made from, unpacked each sdist and ran this script:
```
import os


def get_filelist(root):
    allfiles = []
    for path, subdirs, files in os.walk(root):
        for name in files:
            allfiles.append(os.path.join(path, name)[len(root):])

    return allfiles


path1 = 'numpy_master/numpy-1.17.0.dev0+48b0f3c/'
path2 = 'numpy_manifest/numpy-1.17.0.dev0+a6bc29d/'

files_master = get_filelist(path1)
files_new = get_filelist(path2)

print("Files now in sdist that weren't included previously:")
for name in files_new:
    if not name in files_master:
        print(name)

print("\n\nFiles now left out:")
for name in files_master:
    if not name in files_new:
        print(name)
```

The result:
```
Files now in sdist that weren't included previously:
cythonize.dat
numpy/linalg/lapack_lite/fortran.py
numpy/linalg/lapack_lite/f2c_d_lapack.f.patch
numpy/linalg/lapack_lite/clapack_scrub.py
numpy/linalg/lapack_lite/LICENSE.txt
numpy/linalg/lapack_lite/f2c_s_lapack.f.patch
numpy/linalg/lapack_lite/f2c_config.c.patch
numpy/linalg/lapack_lite/f2c_z_lapack.f.patch
numpy/linalg/lapack_lite/README.rst
numpy/linalg/lapack_lite/f2c_lapack.f.patch
numpy/linalg/lapack_lite/make_lite.py
numpy/linalg/lapack_lite/f2c_c_lapack.f.patch
numpy/linalg/lapack_lite/wrapped_routines


Files now left out:

```

I think those should all be included.

The `LICENSE.txt` file was the important one pointed out in the bug report. The missing `cythonize.dat` was also not good, that must have triggered recompilation of Cython files for source builds that we didn't intend have happen.

### CODE REVIEW

**Comment by eric-wieser** [comment]
I'm not sure any of the `lapack_lite` files need to be included. They represent a manual build step that is performed prior to committing, and are not something anyone else should either be running or depending upon.

File: MANIFEST.in [hunk 1]
```diff
@@ -11,15 +11,18 @@ include site.cfg.example
 include numpy/random/mtrand/generate_mtrand_c.py
 recursive-include numpy/random/mtrand *.pyx *.pxd
 # Add build support that should go in sdist, but not go in bdist/be installed
+# Note that sub-directories that don't have __init__ are apparently not
+# included by 'recursive-include', so list those separately
+recursive-include numpy
 recursive-include numpy/_build_utils *
-recursive-include numpy/linalg/lapack_lite *.c *.h
+recursive-include numpy/linalg/lapack_lite *
```
Comment (eric-wieser) [inline MANIFEST.in]: I think this line should be reverted - excluded the intermediate files which are used only as a manual upgrade-lapack tool made sense to me.

Comment (rgommers) [reply MANIFEST.in]: I think you're thinking that they're not needed for installation, hence don't package? I'm thinking of the sdist more as an archive of the tagged source tree (the first purpose in https://www.python.org/dev/peps/pep-0517/#terminology-and-goals). We also package `benchmarks`, `docs` and `tools`, and I don't see a very strong reason to just leave out these files.

**Comment by rgommers** [comment]
> I'm not sure any of the lapack_lite files need to be included. They represent a manual build step that is performed prior to committing, and are not something anyone else should either be running or depending upon.

I think you're thinking that they're not needed for installation, hence don't package? I'm thinking of the sdist more as an archive of the tagged source tree (the first purpose in https://www.python.org/dev/peps/pep-0517/#terminology-and-goals). We also package `benchmarks`, `docs` and `tools`, and I don't see a very strong reason to leave out only these few files.

**Comment by rgommers** [comment]
Hmm, the travisci build complains: `warning: no files found matching 'cythonize.dat'`

**Comment by mattip** [comment]
"cythonize.dat" is a build artifact containing the hashes of the target files used to trigger a cythonize run when they change.

**Comment by rgommers** [comment]
> "cythonize.dat" is a build artifact containing the hashes of the target files used to trigger a cythonize run when they change.

Yes I know - we should be including it to avoid users who build from source having to recompile the Cython files. I think it fell out of the sdist by accident, but I'm not sure yet - will investigate.

**Comment by rgommers** [comment]
> I think it fell out of the sdist by accident, but I'm not sure yet - will investigate.

Okay, we avoid the need for that because we only call `generate_cython` from within this if-statement: `if not os.path.exists(os.path.join(cwd, 'PKG-INFO')):`. So never for `sdist`s. Removed `cythonize.dat` again, since setuptools is being weird about not finding it (locally that works fine for me, probably some setuptools behavior change).

File: MANIFEST.in (hunk 1 above)
Comment (eric-wieser) [reply MANIFEST.in]: Correct, that was my line of reasoning. I buy your argument that these files are of roughly the same relevance as "tools" though.

**Comment by rgommers** [comment]
Okay all green, should be good to merge

**Comment by mattip** [comment]
Thanks @rgommers

**Pull request merged**